powerhome/playbook
# Improve Rails tooltip kit performance when a lot of tooltip kits are on the page

Fixes #1245 

1. upgrades `@popperjs/core` to version `2.6.0`
2. optimizes trigger event logic
3. deprecates `trigger_element_id` for `trigger_element_selector`
4. replaces show/hide logic with create/destroy Popper lifecycle methods

#### Screens

![tooltips](https://user-images.githubusercontent.com/2293844/102413211-34f6f980-3fba-11eb-804b-1cbe62fea299.gif)

#### Breaking Changes

No. Changes are backwards compatible. Existing kit instances still sending `trigger_element_id` will continue to work as see in the last doc example. 

#### Runway Ticket URL

https://nitro.powerhrg.com/runway/backlog_items/NUXE-350

#### How to test this

1. Go to https://playbook.powerapp.cloud/kits/tooltip
2. ensure that no JS errors occur and that all tooltip examples work
3. visit https://nitro.powerhrg.com/dev_docs/playbook
4. ensure the same as #2 above

#### Checklist:


*The normal release cut off deadline is 3p EDT each week. Please reach out to the release team if you have an urgent request or need a release off cycle.*
